napari/layers/base/_slice.py

```diff
@@ -0,0 +1,8 @@
+from itertools import count
+
+_request_ids = count()
+
+
+def _next_request_id() -> int:
+    """Returns the next ID associated with a slice."""
+    return next(_request_ids)
```

Comment:
how does this work?

Reply:
We discussed over zoom and I added a few more docs here to help clarify.